### PR TITLE
Better Core JS Logic + Table View for All Notes

### PR DESCRIPTION
**Pull request by SivamAgarwalla**
Changes
-Refactored entire popup.js file for better structure and logic flow! Now there are no bugs in syncing the correct values or populating the fields
-Lot more metadata is being stored about the note - the user can change the title/url/text and all of it will instantly update in the table and be synced properly
-All notes are being rendered in a table

### CODE REVIEW

**Review by yesollee1 — APPROVED** [review]

File: src/js/popup.js [hunk 1]
```diff
@@ -1,107 +1,119 @@
...
         }
-    }
+    };
 
-    function getPreviouslySavedNotes() {
-        // https://developer.chrome.com/docs/extensions/reference/storage/
+    const fetchAllPreviousNotes = function () {
         chrome.storage.sync.get(
-            [KEY_NOTES_URLS, CURRENT_URL],
+            ['CHROME_STORAGE_ALL_NOTES_KEY'],
```
Comment (jontsai) [inline src/js/popup.js:89]: this needs to be the constant, not a string

File: src/js/popup.js [hunk 2]
```diff
@@ -1,107 +1,119 @@
...
-                        url +
-                        '</a></li>'
-                    );
-                });
-
-                $('#notes-urls').html(
-                    '<ul class="list-group">' + urlLinks.join('') + '</ul>'
-                );
+                ALL_NOTES_OBJECT = result.CHROME_STORAGE_ALL_NOTES_KEY;
+                console.log(ALL_NOTES_OBJECT);
```
Comment (jontsai) [inline src/js/popup.js:92]: remove `console.log` before submitting pull requests

**Review by jontsai — CHANGES_REQUESTED** [review]

File: src/js/popup.js (hunk 1 above)
Comment (SivamAgarwalla) [reply src/js/popup.js:89]: I tried using a constant and it was not fetching properly, but when I use a string for the call it is working properly.

File: src/js/popup.js (hunk 2 above)
Comment (SivamAgarwalla) [reply src/js/popup.js:92]: Okay noted, will do!

File: src/js/popup.js (hunk 1 above)
Comment (jontsai) [reply src/js/popup.js:89]: Yeah, something isn't working properly, then...

**Review by jontsai — APPROVED** [review]
LFTM

**Pull request closed without merging**